src/CoreBundle/private/library/classes/TFullGroupTable/TFullGroupTable.class.php

```diff
@@ -415,6 +415,61 @@ private function getInlineFromAttributes(array $attributes)
         return $inline;
     }
 
+    private function getTableEditorPagedef($pagedef): string
+    {
+        $inputFilterUtil = $this->getInputFilterUtil();
+        $customTableEditor = $inputFilterUtil->getFilteredInput('sTableEditorPagdef');
+        if (null !== $customTableEditor && '' !== $customTableEditor) {
+            $pagedef = $customTableEditor;
+        }
+        return $pagedef;
+    }
+
+    private function getAjaxRecordsUrl(): string
+    {
+        $pagedef = $this->getTableEditorPagedef('tablemanager');
+        $inputFilterUtil = $this->getInputFilterUtil();
+        $tableId = $inputFilterUtil->getFilteredInput('id');
+        $urlUtil = ServiceLocator::get('chameleon_system_core.util.url');
+        $sAjaxURL = $urlUtil->getArrayAsUrl([
+            'id' => $tableId,
+            'pagedef' => $pagedef,
+            '_rmhist' => 'false',
+            'sOutputMode' => 'Ajax',
+            'module_fnc[contentmodule]' => 'ExecuteAjaxCall',
+            '_fnc' => 'getAutocompleteRecordList',
+            'sRestrictionField' => '',
+            'sRestriction' => '',
+            'recordID' => ''
+        ], PATH_CMS_CONTROLLER.'?', '&');
+
+        return $sAjaxURL;
+    }
+
+    private function switchToSelectedRecord(): string
```

Comment:
Can we somehow move this JS code to a JS file? It's strange to add JS code conditionally - I'd advocate for providing this code always, but block usage in another way where it shouldn't be used.

If we cannot remove it, we should rename it, as it doesn't actually switch but only provides the code needed for switching. How about `GetSwitchToRecordCode()`?

Reply:
The JS code here is not conditional, the problem is that it contains a recordUrl that comes from php.
This function "switchRecord()" is called by the select2 function in table.js, actually to change the record. Hmm, I think the name says what the function does.